sale_delivery_date/models/sale_order_line.py

```diff
@@ -5,176 +5,341 @@
 from datetime import datetime, time, timedelta
 
 import pytz
+from pytz import UTC, timezone
 
-from odoo import api, fields, models
+from odoo import api, models
 
 from odoo.addons.partner_tz.tools import tz_utils
 
 _logger = logging.getLogger(__name__)
 
 
 class SaleOrderLine(models.Model):
+    """This override adds delays to the date_deadline and the date_planned.
+    As per this commit 57f805f71e9357870dfc2498c5ef72ebd8ab7273
+    - On pickings, the date_deadline represents the delivery date, and the
+      date_planned represents the preparation date (date_deadline - security_lead).
+    - On sale orders, date_planned represents the delivery date.
+    """
+
     _inherit = "sale.order.line"
 
+    # =====
+    # OVERRIDES
+    # =====
+
+    def _action_launch_stock_rule(self, previous_product_uom_qty=False):
+        # Add a context key, so we know in `_prepare_procurement_values` that we
+        # might have to update the date based on today instead of the order date
+        if previous_product_uom_qty:
+            self = self.with_context(updated_line_qty=True)
+        return super(SaleOrderLine, self)._action_launch_stock_rule(
+            previous_product_uom_qty
+        )
+
     def _prepare_procurement_values(self, group_id=False):
+        # Here, we need to set date_deadline and date_planned correctly
+        # res["date_planned"] is order.date_order + lead_time
+        # So, date_planned should be:
+        # date_planned - lead, on which we apply the cutoff and then the workload with
+        # respect to the calendar (if any)
+        # Also, date_deadline should be:
+        # date_planned on which we apply the customer's time windows
         res = super()._prepare_procurement_values(group_id=group_id)
-        res = self._cutoff_time_delivery_prepare_procurement_values(res)
-        res = self._warehouse_calendar_prepare_procurement_values(res)
-        res = self._delivery_window_prepare_procurement_values(res)
+        # There's 2 cases here:
+        # 1) commitment date is set, compute date_planned from date_deadline
+        # 2) commitment date isn't set, update date_deadline according to
+        #    cutoff / calendar / delivery window, and compute date_planned
+        if self.order_id.commitment_date:
```

Comment:
As you now depend on sale_order_line_date
```suggestion
        if self.commitment_date or self.order_id.commitment_date:
```

Reply:
<s>Mah, we shouldn't need that, as order's commitment date is computed based on its line's commitment_dates.</s>

I was wrong this is not the case...
I think we should drop the dependency anyway.
@sebalix why did we added this `sale_order_line_date` dep? I'm not sure to understand why, according to this commit. Could we drop it?
https://github.com/OCA/sale-workflow/pull/1967/commits/7b68a2e7d69de05fc12e4d345b74ea25121fc55a

Reply:
Because of this (copy/paste):

> **[FIX] sale_delivery_date: add dependency]**
> 
> Add `sale_order_line_date` as dependency because this addon is overriding the same methods (`_onchange_commitment_dat`e and `_prepare_procurement_values`) and we want to ensure than `sale_delivery_date` takes the precedence over `sale_order_line_date`.
> 
> Also, the other feature of `sale_order_line_date` which sync the commitment_date between SO and its lines is welcome in `sale_delivery_date`.

https://github.com/OCA/sale-workflow/pull/1968

If you think it is not needed anymore, feel free to remove it.